sanssouci/post_hoc_bounds.py

```diff
@@ -39,11 +39,49 @@ def max_fp(p_values, thresholds):
         Annals of Statistics, 48(3), 1281-1303.
     """
 
+    s = len(p_values)
+    if s == 0:
+        return(0)
+
+    all_max_fp = curve_max_fp(p_values, thresholds)
+    return all_max_fp[s - 1]
+
+
+def max_fp_slow(p_values, thresholds):
```

Comment:
Hm, why should we have a "slow" version ? Are there cases where it should be applied ? If yes, we should probably rename it, document the difference between `max_fp` and `max_fp_slow`. And both should be tested.

Reply:
The "slow" version is quadratic in the number of input p-values, whereas the other one is linear.  I find it nice to be able to test against the slow version (which used to be done but now the test in https://github.com/pneuvial/sanssouci.python/blob/main/sanssouci/tests/test_post_hoc_bounds.py#L91 is circular, I should fix this).  An alternative is to just drop the slow version and test against known cases.  Would it make sense to keep the slow one as an internal function?

Reply:
I'd drop the slow version whenever you feel confident enough about the fast one. Because the API of the fast version may evolve across time, making the comparison harder and harder anyhow.